tools/accuracy_checker/accuracy_checker/evaluators/custom_evaluators/asr_encoder_prediction_joint_evaluator.py

```diff
@@ -0,0 +1,456 @@
+        for network_dict in network_list:
+            self._part_by_name[network_dict['name']].load_model(network_dict, launcher)
+
+    def _add_raw_encoder_predictions(self, encoder_prediction):
+        for key, output in encoder_prediction.items():
+            if key not in self._raw_outs:
+                self._raw_outs[key] = []
+            self._raw_outs[key].append(output)
+
+    def get_network(self):
+        return [{'name': 'encoder', 'model': self.encoder.network},
+                {'name': 'prediction', 'model': self.prediction.network},
+                {'name': 'joint', 'model': self.joint.network}]
+
+    def decoder(self, identifiers, logits):
+        output = []
+        raw_outputs = []
+        batches = logits.shape[0]
+        for batch_idx in range(batches):
+            inseq = np.squeeze(logits[batch_idx, :, :])
+            # inseq: TxBxF
+            logitlen = inseq.shape[0]
+            sentence = self._greedy_decode(inseq, logitlen)
+            output.append(sentence)
+            # raw_outputs.append(raw_output)
+        result = self.adapter.process(output, identifiers, [{}])
+
+        return raw_outputs, result
+
+    def _greedy_decode(self, x, out_len):
+        hidden_size = 320
+        hidden = (np.zeros([2, 1, hidden_size]), np.zeros([2, 1, hidden_size]))
+        label = []
+        for time_idx in range(out_len):
+            f = np.expand_dims(np.expand_dims(x[time_idx, ...], 0), 0)
+
+            not_blank = True
+            symbols_added = 0
+
+            while not_blank and symbols_added < self._max_symbols_per_step:
+                g, hidden_prime = self._pred_step(
+                    self._get_last_symb(label),
+                    hidden
+                )
+                hidden_prime = (g['151'], g['152'])
+                g = g['153']
+                logp = self._joint_step(f, g, log_normalize=False)[0, :]
+
+                k = np.argmax(logp)
+
+                if k == self._blank_id:
+                    not_blank = False
+                else:
+                    label.append(k)
+                    hidden = hidden_prime
+                symbols_added += 1
+
+        return label
+
+    def _pred_step(self, label, hidden):
+        if label == self._SOS:
+            label = self._blank_id
+        if label > self._blank_id:
+            label -= 1
+        inputs = {'input.1': [[label,]], '1': hidden[0], '2': hidden[1]}
+        return self.prediction.predict(None, inputs)
+
+    def _joint_step(self, enc, pred, log_normalize=False):
+        inputs = {'0': enc, '1': pred}
+        logits, logits_blob = self.joint.predict(None, inputs)
+        logits = logits_blob[:, 0, 0, :]
+        if not log_normalize:
+            return logits
+
+        # probs = F.log_softmax(logits, dim=len(logits.shape) - 1)
+        # return probs
+
+    def _get_last_symb(self, labels) -> int:
+        return self._SOS if len(labels) == 0 else labels[-1]
```

Comment:
can this variable be named without uppercase?

Reply:
fixed